tests/optim/param/test_images.py

```diff
@@ -242,5 +246,268 @@ def test_laplacianimage_random_forward(self) -> None:
+        )
+
+        size = (224, 224)
+        channels = 3
+        batch = 1
+
+        test_tensor = torch.ones(6, 4, 128, 128)
+        output_tensor = image_param.interpolate_tensor(
+            test_tensor, size, batch, channels
+        )
+
+        self.assertEqual(output_tensor.dim(), 4)
+        self.assertEqual(output_tensor.size(0), batch)
+        self.assertEqual(output_tensor.size(1), channels)
+        self.assertEqual(output_tensor.size(2), size[0])
+        self.assertEqual(output_tensor.size(3), size[1])
+
+    def test_sharedimage_single_shape_hw_forward(self) -> None:
+        if torch.__version__ == "1.2.0":
+            raise unittest.SkipTest(
+                "Skipping SharedImage test due to insufficient Torch version."
+            )
+
+        shared_shapes = (128 // 2, 128 // 2)
+        batch = 6
+        channels = 3
+        size = (224, 224)
+        test_param = lambda: torch.ones(batch, channels, size[0], size[1])  # noqa: E731
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+        test_tensor = image_param.forward()
+
+        self.assertEqual(test_tensor.dim(), 4)
+        self.assertEqual(test_tensor.size(0), batch)
+        self.assertEqual(test_tensor.size(1), channels)
+        self.assertEqual(test_tensor.size(2), size[0])
+        self.assertEqual(test_tensor.size(3), size[1])
+
+    def test_sharedimage_single_shape_chw_forward(self) -> None:
+        if torch.__version__ == "1.2.0":
+            raise unittest.SkipTest(
+                "Skipping SharedImage test due to insufficient Torch version."
+            )
+
+        shared_shapes = (3, 128 // 2, 128 // 2)
+        batch = 6
+        channels = 3
+        size = (224, 224)
+        test_param = lambda: torch.ones(batch, channels, size[0], size[1])  # noqa: E731
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+        test_tensor = image_param.forward()
+
+        self.assertEqual(test_tensor.dim(), 4)
+        self.assertEqual(test_tensor.size(0), batch)
+        self.assertEqual(test_tensor.size(1), channels)
+        self.assertEqual(test_tensor.size(2), size[0])
+        self.assertEqual(test_tensor.size(3), size[1])
+
+    def test_sharedimage_single_shape_bchw_forward(self) -> None:
+        if torch.__version__ == "1.2.0":
+            raise unittest.SkipTest(
+                "Skipping SharedImage test due to insufficient Torch version."
+            )
+
+        shared_shapes = (1, 3, 128 // 2, 128 // 2)
+        batch = 6
+        channels = 3
+        size = (224, 224)
+        test_param = lambda: torch.ones(batch, channels, size[0], size[1])  # noqa: E731
+        image_param = images.SharedImage(
+            shapes=shared_shapes, parameterization=test_param
+        )
+        test_tensor = image_param.forward()
+
+        self.assertEqual(test_tensor.dim(), 4)

```

Comment:
Could we potentially also assert the shapes of `image_param.shared_init` and `image_param.offset` ?

Reply:
I've added asserts for `image_param.shared_init` in the latest commit, and I'll add the `image_param.offset` tomorrow.